src/Features/CSharp/Portable/UseLocalFunction/CSharpUseLocalFunctionCodeFixProvider.cs

```diff
@@ -46,111 +46,128 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
         {
             var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
 
-            var localDeclarationToLambda = new Dictionary<LocalDeclarationStatementSyntax, LambdaExpressionSyntax>();
```

Comment:
IMO this should never have been a dictionary. It wasn't used for dictionary lookup, only iteration. If all you need is iteration, just use a List. The dictionary was just a hacky way to group 2 things together.

Reply:
That seems fine.  Some people (like me) say "there's an association here.  a dictionary is a clear way to express that association".  A list of tuples is equally fine for that purpose if you want :)

Reply:
right - but there's no clear "key" and no clear "value", meaning there's no clear association *from* one thing *to* another, there's only association *between* them

Reply:
That seems fine. Some people (like me) say "there's an association here. a dictionary is a clear way to express that association". A list of tuples is equally fine for that purpose if you want :)